### PR TITLE
Convert app-link element to use typescript.

### PR DESCRIPTION
**Pull request by maffoo**
See #220

### CODE REVIEW

File: client-js/app/elements/app-link.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  @property({type: Array})
+  path: string[];
+
+  isSpecialClick(e): boolean {
+    return (e.ctrlKey || e.metaKey || e.button !== MOUSE_MAIN_BUTTON);
+  }
+
+  @listen('click')
+  onClick(e) {
+    if (this.isSpecialClick(e)) return;
```
Comment (adamcw) [inline client-js/app/elements/app-link.ts:16]: Here you have a one line return without curly braces, on line 25 you include the curly braces.

File: client-js/app/elements/app-link.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    this.fire('app-link-click', {path: this.path});
```
Comment (adamcw) [inline client-js/app/elements/app-link.ts:18]: Is the styleguide here:

``` typescript
this.fire('app-link-click', {path: this.path});
```

or 

``` typescript
this.fire('app-link-click', { path: this.path });
```

**Comment by adamcw** [comment]
Two small comments, else LGTM.

File: client-js/app/elements/app-link.ts (hunk 1 above)
Comment (maffoo) [inline client-js/app/elements/app-link.ts:16]: LOL; I'm inclined to remove curlies below, but how do you feel about these one-line if statements?

File: client-js/app/elements/app-link.ts (hunk 2 above)
Comment (maffoo) [inline client-js/app/elements/app-link.ts:18]: For js at least it's the former: https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting

File: client-js/app/elements/app-link.ts (hunk 1 above)
Comment (adamcw) [inline client-js/app/elements/app-link.ts:16]: I personally prefer braces for consistency, and if you include braces, I prefer not to one-line them.

``` typescript
if (this.isSpecialClick(e)) {
  return;
}
```

Everyone is different on this though.

File: client-js/app/elements/app-link.ts (hunk 2 above)
Comment (adamcw) [inline client-js/app/elements/app-link.ts:18]: Seems to be the same for Python too. This is fine then.

File: client-js/app/elements/app-link.ts (hunk 1 above)
Comment (maffoo) [inline client-js/app/elements/app-link.ts:16]: The js style guide points to the c++ style guide which allows single line conditionals (https://google.github.io/styleguide/cppguide.html#Conditionals), so I went with that. I agree that everyone is different on this; we should set up automated linting/formatting and just use that :-)

**Pull request merged**